BraveShared/Extensions/URLExtensions.swift

```diff
@@ -7,6 +7,19 @@ import Foundation
 import BraveCore
 
 extension URL {
+  /// Attempt to make a url that is sourced from a javascript `URL.href`.
+  ///
+  /// Since javascript's `URL` is more loose with acceptable caracters (such as `|`), we need to ensure such characters are escaped properly before we attempt to make a URL out of it.
+  public init?(javascriptURLString urlString: String) {
+    if let url = URL(string: urlString) {
```

Comment:
Does it create a correct URL if you use `NSURL(idnString:)` instead of `URL(string:)` with the `|` character?

Reply:
It does actually